### PR TITLE
feat: complete fault injection features

### LINKED ISSUE #54: Storage: Introduce failures into uploads with return-XXX-after-YYYk when using Retry Test API
**What component of `google-cloud-cpp` is this feature request for?**
`google/cloud/storage`

**Describe the solution you'd like**
Introduce failures into uploads with return-XXX-after-YYYk when using Retry Test API. It should behave similarly to emulators existing support but should be flexible for values of XXX and YYYk. Where XXX is a HTTP status code and YYYk is a multiple of 256kib

It was marked as a todo since: https://github.com/googleapis/google-cloud-cpp/pull/6611

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #24. Also fixes #54

### CODE REVIEW

File: testbench/database.py [hunk 1]
```diff
@@ -295,26 +293,24 @@ def get_retry_test(self, retry_test_id):
             testbench.error.notfound("Retry Test %s" % retry_test_id, context=None)
         return self.__to_serializeable_retry_test(retry_test)
 
+    def __validate_injected_failure_description(self, failure):
```
Comment (cojenco) [inline testbench/database.py:296]: 👍 Looks good, thanks!

File: tests/test_emulator_retry.py [hunk 2]
```diff
@@ -99,6 +102,184 @@ def test_retry_test_return_error(self):
         )
         self.assertEqual(list_response.status_code, 429, msg=list_response.data)
 
+    def test_retry_test_return_error_after(self):
+        response = self.client.post(
+            "/retry_test",
+            data=json.dumps(
+                {"instructions": {"storage.buckets.list": ["return-429-after-128K"]}}
```
Comment (cojenco) [inline tests/test_emulator_retry.py:109]: I'm still confused. In `handle_retry_test_instruction()` on Line 416, doesn't it only handle `error_after_bytes_matches` when method == "storage.objects.insert"?

Comment (cojenco) [reply tests/test_emulator_retry.py:109]: Ah I see the other `test_retry_test_return_error_after_bytes()` down there 👍

Comment (coryan) [reply tests/test_emulator_retry.py:109]: Do you need me to change something or are we good to go here?

**Review by cojenco — APPROVED** [review]

**Pull request merged**